drivers/gpio/gpio_stm32.c

```diff
@@ -280,51 +426,19 @@ static int gpio_stm32_config(struct device *dev, int access_op,
 	/* figure out if we can map the requested GPIO
 	 * configuration
 	 */
-	map_res = gpio_stm32_flags_to_conf(flags, &pincfg);
-	if (map_res != 0) {
-		err = map_res;
+	err = gpio_stm32_flags_to_conf(flags, &pincfg);
+	if (err != 0) {
 		goto release_lock;
 	}
 
-	if (gpio_stm32_configure(cfg->base, pin, pincfg, 0) != 0) {
-		err = -EIO;
-		goto release_lock;
-	}
+	gpio_stm32_configure(cfg->base, pin, pincfg, 0);
 
-	if (IS_ENABLED(CONFIG_EXTI_STM32) && (flags & GPIO_INT) != 0) {
-
-		if (stm32_exti_set_callback(pin, cfg->port,
-					    gpio_stm32_isr, dev) != 0) {
-			err = -EBUSY;
-			goto release_lock;
+	if ((flags & GPIO_OUTPUT) != 0) {
+		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			gpio_stm32_port_set_bits_raw(dev, BIT(pin));
+		} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+			gpio_stm32_port_clear_bits_raw(dev, BIT(pin));
 		}
```

Comment:
We should set the initial output value before we enable the output. Enabling the output is done by `gpio_stm32_configure` called on line 434.